### PR TITLE
fix: inconsistencies and errors

### PR DESCRIPTION
**Pull request by LEEDASILVA**
`trimatoi_test` :
- the solution was incorrect, some times it passed and sometimes it did not
- added extra cases to test (for the test cases that did not pass)

`pointone_test`:
- function `PointOne` was not being used

### CODE REVIEW

File: solutions/trimatoi.go [hunk 1]
```diff
@@ -1,31 +1,27 @@
...
-func TrimAtoi(s string) int {
-	runes := []rune(s)
-	for i := range s {
-		runes[i] = rune(s[i])
+import (
+	"strconv"
+	"unicode"
+)
+
+func recursion(s string, i int) string {
```
Comment (xpetit) [inline solutions/trimatoi.go:8]: I don't like recursion, especially to filter a string. Two other ways:
```go
s = strings.Map(func(r rune) rune {
	if (r < '0' || r > '9') && r != '-' {
		return -1
	}
	return r
}, s)
```
```go
var s2 string
for _, r := range s {
	if (r >= '0' && r <= '9') || r == '-' {
		s2 += string(r)
	}
}
```

File: solutions/trimatoi.go [hunk 2]
```diff
@@ -1,31 +1,27 @@
...
-	if numbers[0] == '-' {
-		sign = -1
-		i++
-	} else if numbers[0] == '+' {
-		i++
-	}
-	for ; i < len(numbers); i++ {
-		res = res*10 + int(numbers[i]) - '0'
+func TrimAtoi(s string) int {
+	if a, err := strconv.Atoi(recursion(s, len(s)-1)); err == nil {
```
Comment (xpetit) [inline solutions/trimatoi.go:23]: if `strconv.Atoi` fails it returns `0` by default, so you can confidently return the first value and ignore the second (the error)

**Review by xpetit — CHANGES_REQUESTED** [review]

File: solutions/trimatoi.go (hunk 1 above)
Comment (LEEDASILVA) [reply solutions/trimatoi.go:8]: the solutions looks good but the it does not pass a special case int the tests `"sdx1-fa2W3s4"` for example.
so will add `if (r >= '0' && r <= '9') || (r == '-' && s2 == "")`

File: solutions/trimatoi.go (hunk 2 above)
Comment (LEEDASILVA) [reply solutions/trimatoi.go:23]: nice

**Review by xpetit — APPROVED** [review]

**Pull request merged**